### PR TITLE
Update/2.0.0 staking update

### PR DESCRIPTION
**Pull request by aliXsed**
Staking module builds okay and passes its own tests with this update. Staking module is also enabled in the runtime. The whole cennznet build is okay. But as it was the case with the previous module of this branch, runtime tests still failing as the whole branch is still a WIP. However, I suggest we merge the changes at this stage and fix the runtime/cli and etc tests in another step.

### CODE REVIEW

File: crml/staking/src/mock.rs [hunk 1]
```diff
@@ -111,15 +110,19 @@ impl_outer_origin! {
 }
 
 mod staking {
-	pub use crate::Event;
+	// Re-export needed for `impl_outer_event!`.
+	pub use super::super::*;
```
Comment (jordy25519) [inline crml/staking/src/mock.rs:114]: (very small nit) doesn't this still work: `pub use crate::Event;`? I think staking::Event is all that's needed

File: runtime/src/impls.rs [hunk 2]
```diff
@@ -173,39 +173,55 @@ impl<T: prml_generic_asset::Trait> SimpleAssetSystem for SimpleAssetShim<T> {
...
 	fn target() -> Weight {
 		TargetBlockFullness::get() * max()
 	}
 
-	// poc reference implementation.
-	fn fee_multiplier_update(block_weight: Weight, previous: Fixed64) -> Fixed64 {
-		let block_weight = block_weight as f32;
-		let v: f32 = 0.00004;
+	// update based on runtime impl.
+	fn runtime_multiplier_update(fm: Multiplier) -> Multiplier {
```
Comment (jordy25519) [inline runtime/src/impls.rs:199]: these changes are abit hard to review, are they copied from plug?

**Review by jordy25519 — APPROVED** [review]
lgtm!

File: runtime/src/impls.rs (hunk 2 above)
Comment (aliXsed) [reply runtime/src/impls.rs:199]: Yes

File: crml/staking/src/mock.rs (hunk 1 above)
Comment (aliXsed) [reply crml/staking/src/mock.rs:114]: I've made it more like what it is in Plug.

**Pull request merged**